packages/next/src/server/request/params.ts

```diff
@@ -210,7 +211,38 @@ function makeAbortingExoticParams(
 
   const promise = makeHangingPromise<Params>(
     prerenderStore.renderSignal,
-    '`params`'
+    '`params`',
+    {
+      get: function get(target, prop, receiver) {
+        if (prop === 'then' || prop === 'catch' || prop === 'finally') {
+          const originalMethod = ReflectAdapter.get(target, prop, receiver)
+
+          return {
+            [prop]: (...args: unknown[]) => {
+              const store = dynamicAccessAsyncStorage.getStore()
+
+              if (store)
+                store.abortController.abort(
+                  new Error(`Accessed fallback \`params\` during prerendering.`)
+                )
+
+              // Make sure the original method of the proxied promise is called.
+              // We don't use the returned new promise though. This is mostly
+              // needed to ensure the hanging promise rejection is triggered
+              // when the prerender signal is aborted.
+              originalMethod.apply(target, args)
+
+              // Instead, we return the instrumented proxied promise again, to
+              // ensure that it's also used when params.then(), params.catch(),
+              // or params.finally() is passed to another function.
+              return promise
```

Comment:
clever but maybe breaks semantics. Like does `params.finally().catch()` work? I think technically the catch should run if the finally throws. The params itself will reject which would cause the finally to reject but you wouldn't observe that in the final catch b/c in reality you are attaching the catch to the params promise not the finally result. Probably fixable by returning a new Proxy around the promise so you can capture the result of the `originalMethod.apply` call and chain the subsequent calls to that.

But all this proxy code was supposed to be temporary so we can support exotic properties of these APIs but this would be permanent. Probably solveable but I'm hesitant about the incomapatibility here.